### PR TITLE
cln-grpc-plugin: Secure access to your node over the network

### PR DESCRIPTION
**Pull request by cdecker**
This is the final PR in the cln-* series. It uses all the primitives we built in the previous 3 PRs and uses them to expose the JSON-RPC over grpc, with mTLS authentication builtin.

### CODE REVIEW

File: plugins/grpc-plugin/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[package]
+edition = "2021"
+name = "grpc-plugin"
+version = "0.1.0"
```
Comment (vincenzopalazzo) [inline plugins/grpc-plugin/Cargo.toml:4]: maybe it is missing a space here between the two sections?

**Review by vincenzopalazzo — COMMENTED** [review]
I was searching about the plugin architecture, to think about how to implement the `grpc` interface for my dart wrapper. In the meanwhile I left one comment (very very stupid).

Half of the code is the `tls` config, and I'm not the right person to review it

File: plugins/grpc-plugin/Cargo.toml (hunk 1 above)
Comment (cdecker) [reply plugins/grpc-plugin/Cargo.toml:4]: Fixed up in 56d1820c9

**Comment by rustyrussell** [comment]
Should this get a Changelog-Added entry?  Should it be in the default PLUGINS list for builtin plugins if rust is supported?  (Otherwise make install won't install it).

**Comment by cdecker** [comment]
> Should this get a Changelog-Added entry? Should it be in the default PLUGINS list for builtin plugins if rust is supported? (Otherwise make install won't install it).

I'm tempted to mark this as experimental for this release, since I was unable to extensively test the conversion code from JSON-RPC to grpc and back. Shall I wrap the respective `Makefile` entries with experimental markers? Hence the lack of Changelog entries atm.

**Comment by rustyrussell** [comment]
I think experimental is too harsh: if it requires a config option to activate, that's enough. IMHO.

Plugins generally don't have the same requirements as core functionality IMHO.

**Comment by cdecker** [comment]
@rustyrussell I have your branch merged in and it compiles. Will spend today adding more types since they can be difficult to add later, and add some tests to verify the generated conversions are working ok :+1:

File: plugins/grpc-plugin/src/main.rs [hunk 2]
```diff
@@ -31,13 +31,17 @@ async fn main() -> Result<()> {
...
-            options::Value::Integer(50051),
+            options::Value::Integer(-1),
             "Which port should the grpc plugin listen for incoming connections?",
         ))
         .start()
         .await?;
 
     let bind_port = match plugin.option("grpc-port") {
+        Some(options::Value::Integer(-1)) => {
+            log::warn!("`grpc-port` option is not configured, exiting.");
```
Comment (rustyrussell) [inline plugins/grpc-plugin/src/main.rs:42]: This is probably just log::info?  It's a great plugin, but every non-activated plugin should probably not shout to be activated :)

**Review by rustyrussell — CHANGES_REQUESTED** [review]
Trivial change requested... I would merge anyway, but I know more is coming...

**Comment by cdecker** [comment]
Fixed up. The remaining grpc methods are in a separate PR, so feel free to merge this one :-)

**Review by rustyrussell — APPROVED** [review]

**Comment by rustyrussell** [comment]
Pushed a simple test race fix.

Ack 343961d1c430eb5072264276ef096d109f5091a1

File: plugins/grpc-plugin/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+[package]
+edition = "2021"
+name = "cln-grpc-plugin"
+version = "0.1.0"
```
Comment (vincenzopalazzo) [inline plugins/grpc-plugin/Cargo.toml:4]: if it is incompleted (request and response missed) maybe we can use the convention like this one?
```suggestion
version = "0.0.1"
```

And release the `0.1.0` only after all req/resp are supported?

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/5013/commits/343961d1c430eb5072264276ef096d109f5091a1

(with just some small comment!)

File: plugins/grpc-plugin/Cargo.toml (hunk 3 above)
Comment (cdecker) [reply plugins/grpc-plugin/Cargo.toml:4]: Good idea :+1:

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/pull/5013/commits/a47770dd4e7a8ea65d0e915fba25dc8731160797

**Pull request merged**